vcr/signature/proof/legacy.go

```diff
@@ -0,0 +1,75 @@
+package proof
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/crypto/hash"
+	"github.com/nuts-foundation/nuts-node/vcr/signature"
+	"strings"
+	"time"
+)
+
+// LegacyLDProof is a simple/wrong implementation of the LDProof specification. It uses a simple canonicalization
+// algorithm and base64 encodes the payload before signing. It is here for backwards compatibility.
+type LegacyLDProof struct {
+	vc.JSONWebSignature2020Proof
+}
+
+// Sign signs a provided document with the provided key.
+// Deprecated: this method is the initial and wrong implementation of a JSON-LD proof. There will be a new method added in the near future.
+func (p LegacyLDProof) Sign(document map[string]interface{}, suite signature.Suite, key crypto.Key) (interface{}, error) {
+	kid, err := ssi.ParseURI(key.KID())
+	if err != nil {
+		return nil, fmt.Errorf("unable to sign proof: unable parse KID as ssi.URI")
+	}
+
+	p.Type = suite.GetType()
+	p.ProofPurpose = "assertionMethod"
+	p.Created = time.Now()
+	p.VerificationMethod = *kid
+
+	// Don't use the suite's canonicalization method because it messes up the order of the fields:
+	// (this is one of the reasons this proof is deprecated)
+	// canonicalProof, err := suite.GetCanonicalDocument(p.asMap())
+	canonicalProof, err := json.Marshal(p)
+	if err != nil {
+		return nil, err
+	}
+
+	canonicalDocument, err := suite.GetCanonicalDocument(document)
```

Comment:
but then why still canonicalizr the document here? Does it matter?

Reply:
This is the exact copy of the current implementation. Lets not mess with it. The current "canonicalization" step is just a json.Marshall.